.github/scripts/shrink-brew.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Copyright (c) 2019 Kevin R Croft <krcroft@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# This script reduces the size of a brew-managed installation on macOS.
+# Its main use is to shrink this area to fit within GitHub's cache
+# limits however it can also be used by end-users wanting to save space.
+#
+# Note that this script remove alot of content deemed unecessary for our
+# purposes such as ruby, go, grade, azure, etc..) so using on your
+# home system is probably not what you want :-)
+#
+# Usage: ./shrink-brew.sh
```

Comment:
By shrinking brew, we make its size fall within github's cache limit, enabling us to reduce our macos CI runtimes by roughly 7 minutes when we restore from cache instead of installing gcc and deps each time.

If we don't want to use the cache them yes, the shrinkers for brew, macports, and msys2 can all be discarded. 

But even if we want to use the cache to speed up our runs: your point about these specific scripts being non-dosbox specific are interesting and true.  Infact, another user of github's cache mentioned using these to help him too. 

Would it make sense for me to make another repo and put them there? 'github-cache-tools', and then we would pull that in? Curious how best we can do this.

Reply:
Right now, using official tools, macOS builds in ~2:20; it would build even faster if we didn't trigger unnecessary installation of clang compiler via brew (it's preinstalled already), when I had the very first version of macOS builds they were finishing in [~1:08](https://github.com/dreamer/dosbox-staging/runs/237824589). This caching is needed only because new macOS build configurations are introduced…

Few additional warnings on non-default and hard to set up build configuration, using CPU architecture and compiler already covered by other jobs is not worth precious build job slot and all this additional code that needs to be maintained.

I would be fine if we had this outside of our primary CI jobs, on a machine that can be pre-configured for this exact workload.

I think the end goal is to have high quality, developer-friendly, user-friendly DOSBox implementation that is easy to maintain - covering every possible build configuration is not really necessary for that (despite of opinions of some Vogons posters).

Reply:
It's all skipped with the cache; so we're getting two full builds (optimized and debug) done in roughly 2 minutes flat. https://github.com/dreamer/dosbox-staging/runs/311090468

Reply:
The long-pole by a great amount is Clang on Windows, and the code analysis.  Both are about 12 minutes.  Meanwhile, Linux and macOS builds both are done all their builds after just 2 to 3 minutes.  

Personally, I'm unable to update code any faster than that tempo anyway; I'm still the long pole - even if all these jobs took 25+ minutes. 

I do like the idea of quick feedback though - so I want to whittle down the "push" CI set to a minimal core set of jobs, and then have the full set run daily for the branches that have changed during that day.. I'm not sure how to accomplish it though.

Reply:
For now, suggest we keep a directory `dormant` inside workflows where we can keep known-good workflows that we've taken out of the current CI set.  I will move MacPorts there.

Reply:
Fixed in d660ea25